tests/utils/test_orm_event_handlers.py

```diff
@@ -0,0 +1,51 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""These tests ensure compatibility with the deprecation of `sqlalchemy.orm.mapper()` in SQLAlchemy 2.0.0.b1.
+See also: https://docs.sqlalchemy.org/en/21/orm/mapping_styles.html#orm-imperative-mapping
+"""
+from __future__ import annotations
+
+import warnings
+
+import pytest
+from sqlalchemy.exc import InvalidRequestError, SADeprecationWarning
+
+from airflow.utils.orm_event_handlers import setup_event_handlers
+
+
+def test_setup_event_handlers_no_warnings():
+    with warnings.catch_warnings(record=True) as warning_list:
+        try:
+            setup_event_handlers(engine=None)
+        except InvalidRequestError:
+            # We are only interested in the warning that may be issued at the beginning of the method.
+            pass
+
+        # Filter out SADeprecationWarning from the warning list
+        sa_deprecation_warnings = [
+            warning for warning in warning_list if issubclass(warning.category, SADeprecationWarning)
+        ]
+
+        # Assert that no SADeprecationWarning was captured
+        assert (
+            len(sa_deprecation_warnings) == 0
+        ), f"Expected no SADeprecationWarning, but got {len(sa_deprecation_warnings)} warnings."
+
+
+if __name__ == "__main__":
+    pytest.main([__file__])
```

Comment:
```suggestion
```

Reply:
You're absolutely right, I didn't mean to include that part. Will remove.

Reply:
Sometimes this part remaining. That why is review exists.

BTW, just wondering why this is required? IDE can't figure out that this module contain tests?

Reply:
The reason for adding it was so I could debug the test in the IDE, otherwise I have to run the test using the CLI, which is less useful.